doc/sphinx/proof-engine/ltac.rst

```diff
@@ -997,98 +1181,42 @@ Pattern matching on terms: match
       Matches any term with a subterm matching :token:`cpattern`. If there is a match
       and :n:`@ident` is present, it is assigned the "matched
       context", i.e. the initial term where the matched subterm is replaced by a
-      hole. The example below shows how to use such term contexts.
+      hole. The example below shows how to use such term contexts.  Note that `context`
+      (with very similar syntax) appearing after the `=>` is the :tacn:`context` tactic.
 
       For :tacn:`match` and :tacn:`multimatch`, if the evaluation of the :token:`ltac_expr`
       fails, the next matching subterm is tried. If no further subterm matches, the next branch
       is tried.  Matching subterms are considered from top to bottom and from left to
       right (with respect to the raw printing obtained by setting the
-      :flag:`Printing All` flag).
-
-     .. example:: Use of `context` in match on terms
-
-        .. coqtop:: all abort
-
-           Ltac f x :=
-             match x with
-               context f [S ?X] =>
-               idtac X;                    (* To display the evaluation order *)
-               assert (p := eq_refl 1 : X=1);    (* To filter the case X=1 *)
-               let x:= context f[O] in assert (x=O) (* To observe the context *)
-             end.
-           Goal True.
-           f (3+4).
+      :flag:`Printing All` flag).  :ref:`Example<match_term_context_ex>`.
 
    .. todo There's a more realistic example from @JasonGross here:
       https://github.com/coq/coq/pull/12103#discussion_r432996954
 
    .. todo @term after context is actually @lconstr_pattern (term with holes), which is spliced away
       Maybe worth restoring that distinction?
```

Comment:
It' s quite important that you can bind variables in the term after `context`, i.e., that it allows `?X` patterns.